oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/tree/impl/AbstractTree.java

```diff
@@ -150,34 +151,62 @@ protected Iterable<String> getChildNames() {
 
     @Override
     public String toString() {
-        return toString(5);
+        return toJsonString(2);
     }
 
-    private String toString(int childNameCountLimit) {
-        StringBuilder sb = new StringBuilder();
-        sb.append(getPath()).append(": ");
-
-        sb.append('{');
-        for (PropertyState p : getProperties()) {
-            sb.append(' ').append(p).append(',');
+    /**
+     * Represents an OAK tree as a JSON object string
+     * When the representation is limited in depth, {"...":"..."} represents the limit.
+     * Note that, try-catch sentences are used to add "ERROR: "+e.getMessage() values in case of representation error.
+     * Note that, this is a recursive method.
+     * @param depth tree depth to represent. Use -1 for an unlimited depth.
+     * @return json object representation of the tree as a string
+     */
+    public String toJsonString(int depth){
+        if(depth == 0){
+            return quote("...");
         }
-
-        Iterator<String> names = this.getChildNames().iterator();
-        int count = 0;
-        while (names.hasNext() && ++count <= childNameCountLimit) {
-            sb.append(' ').append(names.next()).append(" = { ... },");
+        String str = "{";
+        str += quote("_properties_")+":{ ";
+        for (PropertyState ps : this.getProperties()){
+            str+=quote(ps.getName())+":";
+
+            if(ps.getType().isArray()){
+                str += "[ ";
+                for(int i=0; i<ps.count(); i++){
+                    try {
+                        str += quote(jsonStringEscaper(ps.getValue(Type.STRING, i)))+",";
+                    }catch (Exception e){
+                        str += quote("ERROR:" + jsonStringEscaper(e.getMessage()))+",";
+                    }
+                }
+                str = str.substring(0,str.length()-1); //removing the space or the ,
```

Comment:
it would be` if ( i < ps.count()-1 ) { `at the end of the for and I cannot use it for `for (Tree child : this.getChildren()) {`.
I think it will be more confusing to have two ways of doing the same. Any alternatives?